lib/Sema/TypeCheckDecl.cpp

```diff
@@ -3658,7 +3676,7 @@ bool TypeChecker::isAvailabilitySafeForConformance(
 }
 
 void TypeChecker::typeCheckDecl(Decl *D) {
-  DeclChecker(*this).visit(D);
```

Comment:
`typeCheckDecl` can be static.  That should remove some other type checker dependencies in the constraint system.

Reply:
@CodaFi Yup, I'm planning on sinking all of the `getLegacyGlobalTypeChecker` calls down to the places where we actually access the remaining type checker state.